### ISSUE TITLE
Chat Functionality

### ISSUE THREAD

**Issue by mehtakevan** [samarthhapse/sarthi#11]
I would like to add chat functionality enabled with real-time chating through socket.io library.

**Comment by mehtakevan** [comment]
@sanjay-kv @samarthhapse can you please assign this issue to me

**Comment by IP80808080** [comment]
👋 Hello! I'd be happy to help with this issue. Could you please assign it to me? Additionally, could you please label it for the appropriate level? Thank you! 🚀

**Comment by agxmbedi** [comment]
I am Agamjot singh. Under GSSOC'24, I would like to work on this issue. Pls assign me this!!

**Comment by mehtakevan** [comment]
@samarthhapse Can you just please assign the issue to me.

**Comment by poojaverma9578** [comment]
I would like to make a meaningful contrubution in this project  as I found this aligns with my keen interest,I request you to assign me this .

**Comment by Mandar007007** [comment]
i have a good experience with socket.io library, so please assign this issue to me

**Comment by TomLucasakaTGeek** [comment]
Hello **Mr. @mehtakevan**, as I have worked with a Chat App using **_React.js_** , **_Socket.io and Express.js,_** I would be able to add chat functionality very easily alternatively I can also publish a chat assistant using **IBM Watson Assistant**, assign this issue to me under _GSSOC'24_.

**Comment by mtg718** [comment]
@samarthhapse please assign this issue to me under GSSoc'24.

**Comment by samarthhapse** [comment]
@TomLucasakaTGeek hope you are done with exams . I think I have assigned you with the figma task, don't work on figma design currently. Create a chat functionality till the endpoint and further in end you should be able to integrate in main application codebase

**Comment by TomLucasakaTGeek** [comment]
Thank you @samarthhapse for assigning this issue to me, I will surely complete it asap.

**Comment by samarthhapse** [comment]
yeah! keep building

**Comment by samarthhapse** [comment]
@TomLucasakaTGeek  I have added you as assignee for this issue

**Comment by ayann07** [comment]
Please assign me this task under gssoc'24.

**Comment by samarthhapse** [comment]
I have already assigned this issue

**Comment by samarthhapse** [comment]
@ayann07 wait till 27th I have assigned the login/signup page issue to someone else. If he could not complete it you can create frontend for your own backend. Ow I assign you any new issue

**Comment by samarthhapse** [comment]
@TomLucasakaTGeek keep going

**Comment by samarthhapse** [comment]
@TomLucasakaTGeek is it completed ?

**Comment by TomLucasakaTGeek** [comment]
Yes sir, the UI is more or less done just some minor adjustments and I will raise the PR.

**Comment by Aradhyajha** [comment]
Hello @samarthhapse..
please assign this task to me

**Comment by samarthhapse** [comment]
@TomLucasakaTGeek, Is it ready ?

**Comment by samarthhapse** [comment]
@TomLucasakaTGeek any reply ?

**Comment by TomLucasakaTGeek** [comment]
@samarthhapse I apologise sir, for the delay and for not able to complete the work as expected, my visits to my sick parents, constant tension and reduced focus on this work led to such circumstances. Thank you for being patient with me. I have raised the PR but its not complete.

**Comment by samarthhapse** [comment]
No problem @TomLucasakaTGeek .Take care of parents. Take your own time to complete this functionality. Just complete it 100% and then make PR. This functionality is not urgently required. No worries, take your time and just complete it

**Comment by divyanshu255** [comment]
Assign me this under gssoc'24

**Comment by samarthhapse** [comment]
Hello @TomLucasakaTGeek  completed it totally ?

**Comment by TomLucasakaTGeek** [comment]
Yeah, sorry sir but I am on it. I will probably report back with a new PR in 3-4 days. And then assign me on Meeting functionality, I will surely complete it on time.

**Comment by samarthhapse** [comment]
Okay , done. Keep building !

**Comment by samarthhapse** [comment]
@mehtakevan , @IP80808080 , @divyanshu255 , @agxmbedi , @aasthabagrawal , @Mandar007007 , @mtg718 , @poojaverma9578 

If you are available to work on issue #89 pls comment there

**Comment by ayann07** [comment]
I could have worked on it but I don't have any prior experience working on such issue, any other issue on which I could work? @samarthhapse

**Comment by samarthhapse** [comment]
okay @ayann07 , I will let you know by tonight

**Comment by ayann07** [comment]
Should I work on payment gateway integration issue? I just saw no one is working on it

**Comment by ayann07** [comment]
The last comment it had was 17 days ago

**Comment by ayann07** [comment]
@samarthhapse ?

**Comment by samarthhapse** [comment]
I mean If there is any another issue for you I will comment here

**Comment by samarthhapse** [comment]
@ayann07 can you work on #172 ?

**Comment by Aradhyajha** [comment]
Sir I want to work can u assign it to me

**Comment by samarthhapse** [comment]
@TomLucasakaTGeek pls make PR by tomorrow only. Tomorrow is the last day of last deadline

**Comment by samarthhapse** [comment]
@TomLucasakaTGeek submit today asap

**Comment by TomLucasakaTGeek** [comment]
ok sir

**Comment by samarthhapse** [comment]
pls, share the screenshot , where you are integrating it

**Comment by TomLucasakaTGeek** [comment]
I am sorry sir I was not able to complete it today.

**Comment by samarthhapse** [comment]
Okay, no problem. Take time till tomorrow EOD. But just do it completely

**Comment by samarthhapse** [comment]
@TomLucasakaTGeek , it was the deadline yesterday

**Comment by samarthhapse** [comment]
@TomLucasakaTGeek pls raise PR

**Comment by TomLucasakaTGeek** [comment]
Sir I dont want to cause anymore trouble for you, please unassign me, maybe someone will be able to complete it. I have wasted more than a month already I dont want to waste anymore. Also sir thank you for being so patient with me, your level of patience is absent in current generation.

**Comment by samarthhapse** [comment]
You must inform very early , before starting anything , you have complete Idea that you can implement this or not. I assigned it to you a month before , I thought you were working on it. But you took complete one month and then telling that you can't . It is completely not fair

**Comment by samarthhapse** [comment]
we all know that we can do anything or not, If can't simply tell it . Don't waste time of anyone

**Comment by samarthhapse** [comment]
@mehtakevan , can you complete this issue ?

**Comment by samarthhapse** [comment]
@mehtakevan as yourself have opened this one.

**Comment by samarthhapse** [comment]
@mehtakevan pls comment below

**Comment by agxmbedi** [comment]
@samarthhapse will start working on it.

**Comment by samarthhapse** [comment]
Issue is to simply create a chatbox, where the student and expert can chat with each other. If you are 100% sure that you can complete it by 21 EOD. Pls comment on below.

**Comment by mehtakevan** [comment]
@samarthhapse I would have appreciated receiving it earlier, but I understand. Nevertheless, I can ensure the issue will be completed by 22nd EOD. If this aligns with you, kindly assign the issue to me. And also just clear it once what all I need to implement in this issue.

**Comment by samarthhapse** [comment]
Well ! @mehtakevan . I am assigning this issue

**Comment by samarthhapse** [comment]
Description =>

Issue is to simply create a chat feature , for the individual student and expert to chat with each other

**Comment by samarthhapse** [comment]
@mehtakevan I have assigned it to you. Pls complete it by 22 EOD as you said.

**Comment by samarthhapse** [comment]
@mehtakevan , hope doing well.  Ask any small doubt , if any

**Comment by AbhiDiva96** [comment]
Hello @samarthhapse 
Assigned me

**Comment by samarthhapse** [comment]
Description :
When student sign in and select any expert of his choice , then he will see the profile page of expert. On this page you have to create a chat button. Onclick of this chat button create a chatting feature for student and the expert to chat with each other only

**Comment by samarthhapse** [comment]
@AbhiDiva96  pls try to complete in 2-3 days

**Comment by AbhiDiva96** [comment]
Ok @samarthhapse  got it

**Comment by samarthhapse** [comment]
@AbhiDiva96 started well ? Pls feel free to ask doubts , if any.

**Comment by samarthhapse** [comment]
@AbhiDiva96 , how is it going ?

**Comment by samarthhapse** [comment]
@AbhiDiva96 any doubts ?

**Comment by samarthhapse** [comment]
@AbhiDiva96 , how is it going

**Comment by samarthhapse** [comment]
@AbhiDiva96 any update ?

**Comment by samarthhapse** [comment]
@AbhiDiva96 , pls reply fast

**Comment by AbhiDiva96** [comment]
@samarthhapse  working on it

**Comment by samarthhapse** [comment]
@AbhiDiva96 , hope it's going good

**Comment by samarthhapse** [comment]
@AbhiDiva96 keep updating here and feel free to ask doubt, if any

**Comment by AbhiDiva96** [comment]
@samarthhapse  sure..

**Comment by AbhiDiva96** [comment]
Hey @samarthhapse 
I have done some of the frontend part of chat functionality. 
Sorry to say ...
But due to some urgency I am unavailable to work for next 8 day. 
If you can wait then pls else feel free to assign someone else.
Thankyou so much for your belief on me.

**Comment by jelly-afk** [comment]
@samarthhapse why are there two same issues
i am already working on this

**Comment by samarthhapse** [comment]
@jelly-afk , this issue was opened in earlier phase. A contributor has created only backend in his issue. Hence for frontend I have created another one separately

**Comment by samarthhapse** [comment]
@jelly-afk , I am assigning #182 to you

**Comment by samarthhapse** [comment]
@jelly-afk , don't worry. Keep going and raise PR asap. Mostly tomorrow

**Comment by samarthhapse** [comment]
@jelly-afk , assigned and closing this issue to clear confusions
